src/DataSources/DoctrineDataSource.php

```diff
@@ -0,0 +1,221 @@
+<?php
+
+namespace Mesour\DataGrid;
+
+use Doctrine\ORM\Query;
+use Doctrine\ORM\QueryBuilder;
+use Doctrine\ORM\Tools\Pagination\Paginator;
+
+/**
+ * @author  Martin Procházka <juniwalk@outlook.cz>
+ * @package Mesour DataGrid
+ */
+class DoctrineDataSource implements IDataSource
+{
+    /**
+     * Name of primary column name.
+     * @return string
+     */
+    protected $primary_key = 'id';
+
+    /**
+     * Name of parent column name.
+     * @return string
+     */
+    protected $parent_key = 'parent_id';
+
+    /**
+     * QueryBuilder instance.
+     * @var QueryBuilder
+     */
+    protected $qb;
+
+
+    /**
+     * Initialize Doctrine data source with QueryBuilder instance.
+     * @param QueryBuilder  $qb  Source of data
+     */
+    public function __construct(QueryBuilder $qb)
+    {
+        $this->qb = clone $qb;
+    }
+
+
+    /**
+     * Get total count without applied where and limit.
+     * @return int
+     */
+    public function getTotalCount()
+    {
+        return 0;
```

Comment:
Procházel jsem si Parinator::getCountQuery() a nevidím tam, že by to provádělo Count query bez WHERE, určitě to vrací total count?